src/js/containers/search/table/ResultsTableContainer.jsx

```diff
@@ -116,6 +120,10 @@ export class ResultsTableContainer extends React.Component {
                 this.performSearch();
             }
         }
+        else if (prevProps.columnVisibility !== this.props.columnVisibility) {
+            // Visible columns have changed
+            this.showColumns(this.props.meta.tableType);
```

Comment:
It seems that the container only updates the table UI when it receives new columns from Redux, however it doesn't trigger a new API call. Right now, if you look in the `performSearch` function, the API only receives a limited set of fields (specified in `_requestFields` in the dataMapping file). As you add/remove columns, new searches should be performed to add these fields into the API request. You'll also want to "reset" the search by passing in `true` to the newSearch param in performSearch so that it will reset the table (returning it to the top left), that way if the user is on page 2+, they won't be missing the data for already loaded pages.